docs/spec.md

```diff
@@ -140,11 +140,29 @@ which may allow us to remove tuple types from StableHLO
 
 ```ebnf
 ElementType ::= BooleanType | IntegerType | FloatType | ComplexType
+              | QuantizedType
 BooleanType ::= 'i1'
 IntegerType ::= 'si4' | 'si8' | 'si16' | 'si32' | 'si64'
               | 'ui4' | 'ui8' | 'ui16' | 'ui32' | 'ui64'
 FloatType   ::= 'f8E4M3FN' | 'f8E5M2' | 'bf16' | 'f16' | 'f32' | 'f64'
 ComplexType ::= 'complex' '<' ('f32' | 'f64') '>'
+
+QuantizedType ::= '!quant.uniform' '<' QuantizationStorageTypeSpec
+                ':' QuantizationExpressedType
+                [':' QuantizationDimension]
+                ',' QuantizationScalesAndZeroPoints '>'
+QuantizationStorageTypeSpec ::= QuantizationStorageType
+                              ['<' QuantizationStorageMin ':' QuantizationStorageMax '>']
+QuantizationStorageType ::= IntegerType
+QuantizationStorageMin ::= IntegerConstant
+QuantizationStorageMax ::= IntegerConstant
+QuantizationExpressedType ::= FloatType
+QuantizationDimension ::= IntegerConstant
+QuantizationScalesAndZeroPoints ::= (QuantizationScaleAndZero
+                                  | '{' QuantizationScaleAndZero {',' QuantizationScaleAndZero} '}')
+QuantizationScaleAndZero ::= QuantizationScale ':' QuantizationZeroPoint
+QuantizationScale ::=  FloatConstant
+QuantizationZeroPoint ::=  IntegerConstant
```

Comment:
Proposed rewording:

```ebnf
QuantizedType ::= '!quant.uniform' '<'
                  QuantizationStorageType
                  ['<' QuantizationStorageMin ':' QuantizationStorageMax '>']
                  ':' QuantizationExpressedType
                  [':' QuantizationDimension]
                  ',' QuantizationParameters '>'
QuantizationStorageType ::= IntegerType
QuantizationStorageMin ::= IntegerConstant
QuantizationStorageMax ::= IntegerConstant
QuantizationExpressedType ::= FloatType
QuantizationDimension ::= IntegerConstant
QuantizationParameters ::= QuantizationParameter
                         | '{' QuantizationParameter {',' QuantizationParameter} '}'
QuantizationParameter ::= QuantizationScale ':' QuantizationZeroPoint
QuantizationScale ::= FloatConstant
QuantizationZeroPoint ::=  IntegerConstant
```

Feedback:
  1) No newline needed between ComplexType and QuantizationType.
  2) Aligned some lines similarly to other places in the spec.
  3) Inlined QuantizationStorageTypeSpec - it's just a regular production used only once.
  4) Renamed QuantizationScalesAndZeroPoints to QuantizationParameters.

Reply:
change will be accepted in the next checkin.